modules/transcription-service-microsoft-azure/src/main/java/org/opencastproject/transcription/microsoftazure/MicrosoftAzureTranscriptionService.java

```diff
@@ -280,6 +296,27 @@ public void activate(ComponentContext cc) {
       logger.info("Default '{}' audio encoding will be used", compressedAudioFormat);
     }
 
+    // If language should be detected automatically
+    Option<String> isAutoDetectLanguageOpt = OsgiUtil.getOptCfg(cc.getProperties(), IS_AUTO_DETECT_LANGUAGE);
+    if (isAutoDetectLanguageOpt.isSome()) {
+      defaultIsAutoDetectLanguage = Boolean.valueOf(isAutoDetectLanguageOpt.get());
+    }
+    logger.info("Automatically detecting language is globally enabled: {}", defaultIsAutoDetectLanguage);
+
+    // Which languages should be used in automatic language detection
+    Option<String> autoDetectLanguagesOpt = OsgiUtil.getOptCfg(cc.getProperties(), AUTO_DETECT_LANGUAGES);
+    if (autoDetectLanguagesOpt.isSome()) {
```

Comment:
I would be tempted to hide this behind a check on `defaultIsAutoDetectLanguage` since we don't care about these checks unless the detection is turned on.

Reply:
I think currently the code is trying to accommodate for a scenario where the list of languages is defined in the global config, and the workflow operation later only decides whether automatic detection should be used or not, but does not specify the list of languages (in which case the list of languages from the global config would be used). 

Maybe having the same config in the global config and as a workflow operation key is just a bad idea.

Reply:
No, I just missed the part where it's settable in both.  I can *absolutely* see a use in setting this in both places.  One's a default, one is for handling a hypothetical overseas tenant where they speak an entirely different set of languages.